modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java

```diff
@@ -427,6 +429,16 @@ public Redirect getRedirect(
 
 				actualURL = HttpComponentsUtil.setParameter(
 					actualURL, "doAsUserId", encDoAsUserId);
+
+				if (params.containsKey("doAsUserId")) {
+					Map<String, String[]> modifiedParams = new HashMap<>();
```

Comment:
Hi @AliciaGarciaGarcia , thank you for your suggestion! @adolfopa made a similar suggestion and I went with it since it looked a little bit cleaner to me. Let me know if you disagree.